### PR TITLE
solve issue #130

### PR DESCRIPTION
**Pull request by berndfinger**

### CODE REVIEW

File: tasks/assert-installation.yml [hunk 1]
```diff
@@ -16,10 +16,9 @@
...
     assert:
-      that:
-        - "'{{ line_item|replace(\'@\', \'\') }}' in yum_group_result.stdout_lines"
-      fail_msg: "FAIL: Package group '{{ line_item|replace(\'@\', \'\') }}' is not installed!"
-      success_msg: "PASS: Package group '{{ line_item|replace(\'@\', \'\') }}' is installed."
+      that: "'{{ line_item|replace(\"@\", \"\") }}' in yum_group_result.stdout_lines"
+      fail_msg: "FAIL: Package group '{{ line_item|replace(\"@\", \"\") }}' is not installed!"
+      success_msg: "PASS: Package group '{{ line_item|replace(\"@\", \"\") }}' is installed."
     with_items:
       - "{{ __sap_preconfigure_packagegroups }}"
```
Comment (richm) [inline tasks/assert-installation.yml:23]:
```yaml
  - name: Assert that all required RHEL 7 package groups are installed
    assert:
      that: line_item in yum_group_result.stdout_lines
      fail_msg: "FAIL: Package group '{{ line_item }}' is not installed!"
      success_msg: "PASS: Package group '{{ line_item }}' is installed."
    loop: "{{ __sap_preconfigure_packagegroups | map('replace', '@', '') | list }}"
```

Comment (berndfinger) [reply tasks/assert-installation.yml:23]: I am getting the following output when running the role with the suggested change:
TASK [sap-preconfigure : Assert that all required RHEL 7 package groups are installed] ***********************************************************************************************************************************
skipping: [p81] => (item=[) 
skipping: [p81] => (item=') 
skipping: [p81] => (item=) 
skipping: [p81] => (item=s) 
skipping: [p81] => (item=e) 
skipping: [p81] => (item=r) 
skipping: [p81] => (item=v) 
skipping: [p81] => (item=e) 
skipping: [p81] => (item=r) 
skipping: [p81] => (item=') 
skipping: [p81] => (item=]) 

I tested on Fedora 33 with ansible 2.9.18 against a managed node running RHEL 8.1. With a RHEL 7.6 managed node (for which the task will be executed), the output is different but also shows that the string is converted into separate characters.

Extract:
TASK [sap-preconfigure : Assert that all required RHEL 7 package groups are installed] ***********************************************************************************************************************************
failed: [p76] (item=[) => {
    "ansible_loop_var": "line_item",
    "assertion": "line_item in yum_group_result.stdout_lines",
    "changed": false,
    "evaluated_to": false,
    "line_item": "[",
    "msg": "FAIL: Package group '[' is not installed!"
}
failed: [p76] (item=') => {
    "ansible_loop_var": "line_item",
    "assertion": "line_item in yum_group_result.stdout_lines",
    "changed": false,
    "evaluated_to": false,
    "line_item": "'",
    "msg": "FAIL: Package group ''' is not installed!"
}
failed: [p76] (item=) => {
    "ansible_loop_var": "line_item",
    "assertion": "line_item in yum_group_result.stdout_lines",
    "changed": false,
    "evaluated_to": false,
    "line_item": "",
    "msg": "FAIL: Package group '' is not installed!"
}
failed: [p76] (item=b) => {
    "ansible_loop_var": "line_item",
    "assertion": "line_item in yum_group_result.stdout_lines",
    "changed": false,
    "evaluated_to": false,
    "line_item": "b",
    "msg": "FAIL: Package group 'b' is not installed!"
}
failed: [p76] (item=a) => {
    "ansible_loop_var": "line_item",
    "assertion": "line_item in yum_group_result.stdout_lines",
    "changed": false,
    "evaluated_to": false,
    "line_item": "a",
    "msg": "FAIL: Package group 'a' is not installed!"
}


Here is the full code of the task:
  - name: Assert that all required RHEL 7 package groups are installed
    assert:
      that: line_item in yum_group_result.stdout_lines
      fail_msg: "FAIL: Package group '{{ line_item }}' is not installed!"
      success_msg: "PASS: Package group '{{ line_item }}' is installed."
    loop: "{{ __sap_preconfigure_packagegroups | map('replace', '@', '') | list }}"
    loop_control:
      loop_var: line_item
    ignore_errors: "{{ sap_preconfigure_assert_ignore_errors|d(false) }}"

Comment (richm) [reply tasks/assert-installation.yml:23]: hmm - try this:
```yaml
- name: test
  hosts: localhost
  tasks:
    - set_fact:
        __sap_preconfigure_packagegroups:
          - "@group@a@"
          - "@group@b@"
          - "groupc"
          - "@group@e@"
        stdout_lines:
          - groupa
          - groupb
          - groupc
          - groupc
    - assert:
        that: item in stdout_lines
      loop: "{{ __sap_preconfigure_packagegroups | map('replace', '@', '') | list }}"
```

run with `ansible-playbook -vv the-playbook.yml`

Comment (berndfinger) [reply tasks/assert-installation.yml:23]: It looks like the issue is related to the way how the variables are defined (using a vars file which contains lookup for assigning the variable from another variable, using string concatenation and the string() filter). I am getting the same error when using:
```YAML
- name: test
  hosts: localhost
  vars_files:
    - pb-vars.yml
  tasks:
    - set_fact:
        stdout_lines:
          - groupa
          - groupb
          - groupc
          - groupc
    - assert:
        that: item in stdout_lines
      loop: "{{ __sap_preconfigure_packagegroups | map('replace', '@', '') | list }}"
```
with file pb-vars.yml as follows: 
```YAML
__sap_preconfigure_packagegroups_x86_64:
  - "@group@a@"
  - "@group@b@"
  - "groupc"
  - "@group@e@"

__sap_preconfigure_packagegroups: "{{ lookup('vars','__sap_preconfigure_packagegroups_' + ansible_architecture|string ) }}"
```
After removing the string filter, the code works as expected. Also in the role. However, there are other places in the role and in other roles which I would like to modify similarly. When doing this, we risk to introduce other issues. So I suggest that we keep the code as I proposed and use the new code throughout all roles in branch bz1928398.

**Review by richm — APPROVED** [review]
lgtm

**Pull request merged**